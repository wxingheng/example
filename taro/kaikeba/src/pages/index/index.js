/* eslint-disable jsx-quotes */
import Taro, { Component } from "@tarojs/taro";
import { View, Input, Button } from "@tarojs/components";
import { AtSwipeAction, AtIcon } from "taro-ui";
import "./index.scss";
// import { Mcheckbox } from "./../../components/MCheckbox/index.js";
// eslint-disable-next-line import/first
import Mcheckbox from "@/components/Mcheckbox";
import { connect } from '@tarojs/redux';
import { login, getCollection, removeRecordItem } from '../../actions/user'


@connect(({ user }) => ({
  user
}), (dispatch) => ({
  // login(callback) {
  //   dispatch(login(callback))
  // },
  // getCollection(list) {
  //   dispatch(getCollection(list))
  // },
  // removeRecordItem(id, callback = () => { }) {
  //   dispatch(removeRecordItem(id))
  //   callback()
  // }
}))
export default class Index extends Component {
  config = {
    navigationBarTitleText: "待办"
  };

  constructor(props) {
    super(props);
    this.state = {
      val: "",
      todos: [], // Taro.getStorageSync("todos") || [],
      // show add button
      showAdd: true,
      // add input is focus
      addFocus: false
      // todos: [{ title: "吃饭饭", done: false }, { title: "写代码", done: true }]
    };
    console.log(Taro.getEnv());
  }

  componentWillMount() {
    this.login();
    console.log('11111111111', this.props)
  }

  componentDidShow = () => {
    this.login();
    console.log('22222222')
  }

  async getSetting() {
    const setting = await Taro.getSetting();
    console.log("setting----->", setting);
  }

  async login() {
    let userInfo, loginRes;
    try {
      userInfo = await Taro.getUserInfo();
      loginRes = await Taro.login();
    } catch (err) {
      console.log('err---->', err);
      Taro.switchTab({
        url: '../mine/index'
      });
      return;
    }
    const loginPost = {
      code: loginRes.code,
      rawData: userInfo.rawData,
      signature: userInfo.signature,
      encryptedData: userInfo.encryptedData
    };
    this.loginServer(loginPost);
  }

  async openSetting() {
    Taro.openSetting().then(d => {
      console.log(d);
    });
  }


  // 请求server登录成功
  async loginServer(data) {
    const result = await Taro.ajax({
      url: "/api/login",
      method: "post",
      data
    });
    const {
      data: { token }
    } = result;
    Taro.setStorage({
      key: "token",
      data: token
    });
    // 拉取列表
    this.getData();
  }

  async getData() {
    const result = await Taro.ajax({
      url: "/api/lists"
    });
    if (result.code === -1) {
      Taro.showToast({
        title: result.msg,
        icon: "none",
        duration: 2000
      });
    } else {
      this.setState({
        todos: result.data.rows
      });
    }
  }

  async createItem(data) {
    const result = await Taro.ajax({
      url: "/api/list",
      method: "post",
      data
    });
    console.log("result", result);
  }

  async listSync() {
    const result = await Taro.ajax({
      url: "/api/lists",
      method: "post",
      data: this.state.todos
    });
    console.log("listSync--->", result);
    this.getData();
  }

  save = () => {
    Taro.setStorageSync("todos", this.state.todos);
    // this.listSync();
  };

  handleInout = e => {
    this.setState({
      val: e.detail.value
    });
  };

  handleClick = () => {
    console.log("handleClick");
    this.setState(
      {
        todos: [...this.state.todos, { title: this.state.val, done: false }],
        val: ""
      },
      this.save
    );
  };

  handleChange = (e, i) => {
    console.log(e);
    console.log(i);
    this.setState(
      {
        todos: (() => {
          const todos = [...this.state.todos];
          todos[i]["done"] = e.detail.value;
          return todos;
        })()
      },
      this.save
    );
  };

  handleClear = () => {
    Taro.showLoading({
      title: "清理中"
    });
    setTimeout(() => {
      this.setState(
        {
          todos: this.state.todos.filter(v => !v.done)
        },
        this.save
      );
      Taro.hideLoading();
    }, 2000);
  };

  handleSingle = i => {
    console.log(i);
  };
  handleItem = i => {
    console.log(i);
  };
  handleAdd = () => {
    console.log("handleAdd");
    this.setState({
      showAdd: !this.state.showAdd,
      addFocus: true
    });
  };
  async handleAddBlur(e) {
    console.log("handleAddBlur--->>>", e.target.value);
    const result = e.target.value;
    if (result && result.trim()) {
      const r = await Taro.ajax({
        url: "/api/lists",
        method: "post",
        data: [{ title: result.trim(), done: false }]
      });
      console.log("listSync--->", r);
      this.getData();
      this.setState(
        {
          todos: [...this.state.todos, { title: result.trim(), done: false }],
          showAdd: true,
          addFocus: false,
          val: ""
        },
        this.save
      );
    } else {
      this.setState({
        showAdd: true,
        addFocus: false
      });
    }
  }

  handleCheckboxChange = (i, done) => {
    const todos = [...this.state.todos];
    todos[i]["done"] = !done;
    this.setState(
      {
        todos
      },
      this.save
    );
  };

  async handleItemBlur(e, i) {
    console.log("+++++++++++>>>", e.target.value);
    const todos = [...this.state.todos];
    todos[i]["title"] = e.target.value;
    const result = await Taro.ajax({
      url: "/api/lists",
      method: "post",
      data: [todos[i]]
    });
    console.log("listSync--->", result);
    this.getData();
    this.setState(
      {
        todos
      },
      this.save
    );
  }

  async handleSwipeClick(i) {
    console.log("handleSwipeClick");
    let todos = this.state.todos;
    const item = this.state.todos[i];
    if (item.id) {
      const result = await Taro.ajax({
        url: "/api/list/delete",
        method: "post",
        data: { id: item.id }
      });

      console.log("delete---->>>", result);

      todos.splice(i, 1);
      this.setState(
        {
          todos
        },
        this.save
      );
    } else {
      todos.splice(i, 1);
      this.setState(
        {
          todos
        },
        this.save
      );
    }
  }

  render() {
    console.log('this.props---->', this.props)
    const { todos, showAdd, val, addFocus } = this.state;
    console.log("this.state---->", this.state);
    return (
      <View>
        <View className="header">
          {/**
     <View className="header_left">编辑</View>
          <View className="header_center">清单</View>
    */}
          <View className="header_right" onClick={this.listSync}>
            同步
          </View>
        </View>
        <View className="main">
          <View className="ul">
            {todos.map((item, i) => (
              <AtSwipeAction
                onClick={() => this.handleSwipeClick(i)}
                key={i}
                options={[
                  {
                    text: "删除",
                    style: {
                      backgroundColor: "#FF4949"
                    }
                  }
                ]}
              >
                <View className="li">
                  <Mcheckbox
                    checked={item.done}
                    onClick={() => {
                      this.handleCheckboxChange(i, item.done);
                    }}
                  />

                  <Input
                    className={`input ${item.done && "line-through"}`}
                    disabled={item.done ? true : null}
                    onBlur={e => this.handleItemBlur(e, i)}
                    type="text"
                    value={item.title}
                  />
                </View>
              </AtSwipeAction>
            ))}
          </View>

          {showAdd ? (
            <View className="add" onClick={this.handleAdd}>
              <AtIcon value="add" size="20" color="#bbbbbb" />
            </View>
          ) : (
              // 只能第一次自动获取焦点
              <View className="ul">
                <View className="li">
                  <Mcheckbox />
                  <Input className="input" focus={addFocus} autoFocus type="text" value={val} onBlur={this.handleAddBlur} />
                </View>
              </View>
            )}
        </View>
        <Button onClick={this.openSetting}>设置权限</Button>
      </View>
    );
  }
}
