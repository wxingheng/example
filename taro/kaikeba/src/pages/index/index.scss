.kkdone{
    color: red;
    text-decoration: line-through;
}
.header{
  color: #66a5da;
  font-size: 30px;
  display: flex;
  padding: 16px 22px;
  border-bottom: 1px solid #ccc;
}
.header_left, .header_right, .header_center{
  flex: 1;
}
.header_right{
  text-align: right;
}
.header_center{
  text-align: center;
  color: #000000;
}
.main{
  font-size: 32px;
}
.ul{
  padding: 2px 0 2px 34px;
}
.li{
  display: flex;
  padding: 22px 0;
  border-bottom: 1px solid #ccc;
  &:last-child{
    border: none;
  }
}
.checkbox{
  // flex: 1;
  width: 50px;
  margin-left: 12px;

  // margin-top: -12px;
}
.input{
  flex: 1;
  // padding-top: 16px;
  // margin-left: 14px;
}
.line-through{
  // text-decoration: line-through;
  color: #cccccc
}
.add{
  // border: 1px solid red;
  font-size: 70px;
  color: #bbb;
  line-height: 70px;
  padding-left: 26px;
  min-height: 300px;
}
