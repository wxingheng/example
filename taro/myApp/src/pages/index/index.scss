.index{
    width: 375px;
    background: red;
}