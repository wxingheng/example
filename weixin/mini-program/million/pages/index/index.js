const config = require('./config');
const util = require('../../utils/util.js')
const Dialog = require('../../dist/dialog/dialog');
const wxCharts = require('../../utils/wxcharts.js');
const moment = require('../../utils/moment.js');
var lineChart = null;


//index.js
//获取应用实例
const app = getApp()

Page({
    data: {
        motto: 'Hello World',
        userInfo: {},
        hasUserInfo: false,
        canIUse: wx.canIUse('button.open-type.getUserInfo'),
        test001: 100000,
        test002: 4000,
        test003: 5,
        test004: 1000000,
        test005: 0,
    },
    //事件处理函数
    bindViewTap: function () {
        wx.navigateTo({
            url: '../logs/logs'
        })
    },
    clearInput() {
        const viewData = util.coreComput(parseFloat(this.data.test001), parseFloat(this.data.test002), parseFloat(this.data.test003), parseFloat(this.data.test004), parseFloat(this.data.test005));
        // Dialog({
        //     message: `我与一百万的距离：${viewData.result}`,
        //     selector: '#zan-no-title-dialog'
        // }).then(() => {
        //     console.log('=== dialog ===', 'type: confirm');
        // });

        var series = [{
            name: '我的钱',
            data: viewData.data,
            format: function (val, name) {
                return val.toFixed(2) + '元';
            }
        }];
        lineChart.updateData({
            categories: viewData.categories,
            series: series
        });

        wx.setStorageSync('countResult', viewData)
        wx.navigateTo({
            url: '../result/result'
        })

        // this.setData({
        //     value: ''
        // });
    },
    handleTest001Change(event) {
        this.setData({test001: event.detail.detail.value})

    },
    handleTest002Change(event) {
        this.setData({test002: event.detail.detail.value})

    },
    handleTest003Change(event) {
        this.setData({test003: event.detail.detail.value})
    },
    handleTest004Change(event) {
        this.setData({test004: event.detail.detail.value})
    },
    handleTest005Change(event) {
        this.setData({test005: event.detail.detail.value})
    },
    touchHandler: function (e) {
        lineChart.scrollStart(e);
    },
    moveHandler: function (e) {
        lineChart.scroll(e);
    },
    touchEndHandler: function (e) {
        lineChart.scrollEnd(e);
        lineChart.showToolTip(e, {
            format: function (item, category) {
                return category + ' ' + item.name + ':' + item.data
            }
        });
    },
    createSimulationData: function () {
        var categories = [];
        var data = [];
        for (var i = 0; i < 10; i++) {
            categories.push('201620162-' + (i + 1));
            data.push(Math.random() * (20 - 10) + 10);
        }
        return {
            categories: categories,
            data: data
        }
    },
    onLoad: function () {
        if (app.globalData.userInfo) {
            this.setData({
                userInfo: app.globalData.userInfo,
                hasUserInfo: true
            })
        } else if (this.data.canIUse) {
            // 由于 getUserInfo 是网络请求，可能会在 Page.onLoad 之后才返回
            // 所以此处加入 callback 以防止这种情况
            app.userInfoReadyCallback = res => {
                this.setData({
                    userInfo: res.userInfo,
                    hasUserInfo: true
                })
            }
        } else {
            // 在没有 open-type=getUserInfo 版本的兼容处理
            wx.getUserInfo({
                success: res => {
                    app.globalData.userInfo = res.userInfo
                    this.setData({
                        userInfo: res.userInfo,
                        hasUserInfo: true
                    })
                }
            })
        }

        var windowWidth = 320;
        try {
            var res = wx.getSystemInfoSync();
            windowWidth = res.windowWidth;
        } catch (e) {
            console.error('getSystemInfoSync failed!');
        }

        var simulationData = this.createSimulationData();
        lineChart = new wxCharts({
            canvasId: 'lineCanvas',
            type: 'line',
            categories: simulationData.categories,
            animation: false,
            series: [{
                name: '我的钱',
                data: simulationData.data,
                format: function (val, name) {
                    return val.toFixed(2) + '元';
                    // return val.toFixed(2);
                }
            }],
            xAxis: {
                disableGrid: false
            },
            yAxis: {
                title: '总存款 (元)',
                format: function (val) {
                    return val.toFixed(2);
                },
                min: 0
            },
            width: windowWidth,
            height: 200,
            dataLabel: false,
            dataPointShape: true,
            enableScroll: true,
            extra: {
                lineStyle: 'curve'
            }
        });
    },
    getUserInfo: function (e) {
        app.globalData.userInfo = e.detail.userInfo
        this.setData({
            userInfo: e.detail.userInfo,
            hasUserInfo: true
        })
    }
})
