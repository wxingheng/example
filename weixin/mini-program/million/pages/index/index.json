{
  "navigationBarTitleText": "我的一百万",
  "usingComponents": {
    "zan-button": "../../dist/btn/index",
    "zan-button-group": "../../dist/btn-group/index",
    "zan-cell-group": "../../dist/cell-group/index",
    "zan-field": "../../dist/field/index",
    "zan-panel": "../../dist/panel/index",
    "doc-page": "../../components/doc-page/index",
    "zan-dialog": "../../dist/dialog/index",
    "zan-toast": "../../dist/toast/index"
  }
}
