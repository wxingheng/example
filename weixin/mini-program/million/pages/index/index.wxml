<!--index.wxml-->


<!--<doc-page title="一百万的距离" without-padding>-->
    <!--<view class="container">-->
        <!--<view class="userinfo">-->
            <!--<button wx:if="{{!hasUserInfo && canIUse}}" open-type="getUserInfo" bindgetuserinfo="getUserInfo"> 获取头像昵称-->
            <!--</button>-->
            <!--<block wx:else>-->
                <!--<image bindtap="bindViewTap" class="userinfo-avatar" src="{{userInfo.avatarUrl}}" mode="cover"></image>-->
                <!--<text class="userinfo-nickname">{{userInfo.nickName}}</text>-->
            <!--</block>-->
        <!--</view>-->
    <!--</view>-->

    <!-- Field 基础用法 -->
    <!--<zan-panel title="输入一下">-->
        <zan-cell-group>
            <zan-field
                    title="当前存款"
                    placeholder="请输入"
                    value="{{ test001 }}"
                    bind:change="handleTest001Change"
                    inputType="number"
                    result="元"
                    right=true
            >
            </zan-field>
        </zan-cell-group>
        <zan-cell-group>
            <zan-field
                    title="月存款"
                    placeholder="请输入"
                    value="{{ test002 }}"
                    bind:change="handleTest002Change"
                    inputType="number"
                    result="元"
                    right=true
            >
            </zan-field>
        </zan-cell-group>
        <zan-cell-group>
            <zan-field
                    title="估算年收益利率"
                    placeholder="请输入"
                    value="{{ test003 }}"
                    bind:change="handleTest003Change"
                    inputType="number"
                    result="%"
                    right=true
            >
            </zan-field>
        </zan-cell-group>
        <!--<zan-cell-group>-->
            <!--<zan-field-->
                    <!--title="估算年薪涨幅"-->
                    <!--placeholder="请输入"-->
                    <!--value="{{ test005 }}"-->
                    <!--bind:change="handleTest005Change"-->
                    <!--inputType="number"-->
                    <!--result="%"-->
                    <!--right=true-->
            <!--&gt;-->
            <!--</zan-field>-->
        <!--</zan-cell-group>-->
        <zan-cell-group>
            <zan-field
                    title="小目标"
                    placeholder="请输入"
                    value="{{ test004 }}"
                    bind:change="handleTest004Change"
                    inputType="number"
                    result="元"
                    right=true
            >
            </zan-field>
        </zan-cell-group>
        <zan-button-group>
            <zan-button type="primary" bind:btnclick="clearInput">开始计算</zan-button>
        </zan-button-group>
    <!--</zan-panel>-->



    <!--<view class="">-->
        <!--<canvas canvas-id="lineCanvas" disable-scroll="true" class="canvas" bindtouchstart="touchHandler" bindtouchmove="moveHandler" bindtouchend="touchEndHandler"></canvas>-->
        <!--<view style="margin-top:30px;text-align:center">可以拖动图表哦</view>-->
    <!--</view>-->


    <zan-dialog id="zan-no-title-dialog"></zan-dialog>



<!--</doc-page>-->
