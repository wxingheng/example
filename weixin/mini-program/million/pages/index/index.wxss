/**index.wxss**/
.userinfo {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.userinfo-avatar {
  width: 128rpx;
  height: 128rpx;
  margin: 20rpx;
  border-radius: 50%;
}

.userinfo-nickname {
  color: #aaa;
}

.usermotto {
  margin-top: 200px;
}
.field__title--radius {
  padding-bottom: 10px;
}

.popup-field-example--bottom {
  width: 100%;
  height: 150px;
}

.picker-view-example {
  height: 120px;
}

.picker-view-column-example {
  line-height: 50px;
  text-align: center;
}
