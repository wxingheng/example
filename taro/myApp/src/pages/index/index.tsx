import Taro, { Component, Config } from "@tarojs/taro";
import { View, Text, Button } from "@tarojs/components";
import { Tab } from "./../../components/tab/tab";
import { Clock } from "./../../components/clock/clock";
import { connect } from "@tarojs/redux";
import "./index.scss";

import { add, minus, asyncAdd } from "../../actions/counter";

@connect(({ counter }) => ({
  counter
}), (dispatch) => ({
  add () {
    dispatch(add())
  },
  dec () {
    dispatch(minus())
  },
  asyncAdd () {
    dispatch(asyncAdd())
  }
}))
export default class Index extends Component {
  /**
   * 指定config的类型声明为: Taro.Config
   *
   * 由于 typescript 对于 object 类型推导只能推出 Key 的基本类型
   * 对于像 navigationBarTextStyle: 'black' 这样的推导出的类型是 string
   * 提示和声明 navigationBarTextStyle: 'black' | 'white' 类型冲突, 需要显示声明类型
   */
  config: Config = {
    navigationBarTitleText: "首页"
  };

  componentWillMount() {}

  componentDidMount() {}

  componentWillUnmount() {}

  componentDidShow() {}

  componentDidHide() {}

  onPullDownRefresh() {
    console.log("onPullDownRefresh--->>>");
  }
  navigateTo() {
    // 跳转到目的页面，打开新页面
    Taro.navigateTo({
      url: "/pages/test001/index"
    });
  }

  redirectTo() {
    // 跳转到目的页面，在当前页面打开
    Taro.redirectTo({
      url: "/pages/test001/index"
    });
  }

  render() {
    return (
      <View className="index">
        <Text>Hello world!122223000</Text>
        <Tab />
        <Button onClick={this.navigateTo}>navigateTo 跳转</Button>
        <Button onClick={this.redirectTo}>redirectTo 跳转</Button>
        <Clock />
        <View>{this.props.counter.num}</View>
        <Button className='add_btn' onClick={this.props.add}>+</Button>
        <Button className='dec_btn' onClick={this.props.dec}>-</Button>
        <Button className='dec_btn' onClick={this.props.asyncAdd}>async</Button>
      </View>
    );
  }
}
