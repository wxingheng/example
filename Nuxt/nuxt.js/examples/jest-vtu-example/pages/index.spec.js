import { shallowMount } from '@vue/test-utils'
import index from './index'

describe('index', () => {
  test('mounts properly', () => {
    const wrapper = shallowMount(index)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = shallowMount(index)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
