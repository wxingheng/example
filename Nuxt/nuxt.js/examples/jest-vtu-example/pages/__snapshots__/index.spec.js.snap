// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders properly 1`] = `
<section class="container">
  <div>
    <h1 class="title">
      jest-vtu-example
    </h1>
    <btn-stub label="click me!"></btn-stub>
  </div>
</section>
`;
